cns/restserver/ipam.go

```diff
@@ -790,6 +791,12 @@ func (service *HTTPRestService) AssignAvailableIPConfigs(podInfo cns.PodInfo) ([
 	if numOfNCs == 0 {
 		return nil, ErrNoNCs
 	}
+	// slice to store NCIDs
+	ncIDs := make([]string, 0)
```

Comment:
Why not preallocate the slice with the length as `numOfNCs` ? More over can we use `service.state.ContainerStatus` directly in the loop below ?

Reply:
Changed the allocation size to number of NCs. We are already looping over Container Status not sure what you meant by that part of the comment

Reply:
We can use the `numOfNCs` for the length of the slice ? i.e `make([]string, numOfNCs)`.  Added this because the performance(speed and memory i/o) with predefined length is better than dynamic size of the slice. (https://dev.to/jonathanlawhh/golang-slice-vs-array-and-should-i-declare-my-slice-size-4kng)

For the looping part, I meant why didn't we do this (this would eliminate the variables/slice):
```
...
		for ncId := range service.state.ContainerStatus {
		...
		}

...
```

Reply:
Made an update based on these comments

Reply:
Discussed offline and removed this code to facilitate iteration directly where ncIDs are required